Updating Dependecies, Vessel -> MOPS, fixing CI

Gekctek opened the pull request:
I have been on a journey to update dependencies with breaking changes such as CBOR. Doesn't seem like MOPS handles version conflicts very well so there are a few libraries i need to update, like this one, to get things up to date

Not sure your normal process and im assuming MOPS publish is manual

Gekctek committed:
Updating cbor and sha2 dependecies

nomeata commented:
It's been ages since I touched this code, or MOPS etc. Does someone want to take over maintenance?

Gekctek commented:
I can if you want, my principal is 7357l-qewss-qxuo3-ccvfm-6ofer-bameo-jmkvh-kzlxv-vs4w7-zauoh-jae on MOPS, you can add me as a maintainer of the package
I was thinking it could use an update for tests and whatnot with new MOPS features

Gekctek committed:
Converted tests to use MOPS
archived broken nix (due to issues with new MOPS version)
Removed vessel
Reformatted files

Gekctek committed:
Fixing gen-tests

Gekctek committed:
Fixing ci.yml

Gekctek committed:
Fixing check-strict

Gekctek committed:
Fixing broken ci.yml

Gekctek committed:
Adding wasmtime tool to MOPS for gen-tests

Gekctek committed:
Merge pull request #1 from Gekctek/MOPSTests

Converting Vessel to MOPS and fixing CI

Gekctek commented:
Alright, sounds good
I went ahead and just updated all dependencies, killed vessel, moved tests to use MOPS to bring it up to date with things

I did have a problem with the nix though, so I threw it in a folder to maybe fix later, but there seems to be no CI dependency on it and with MOPS the testing is much easier now. It didn't like the new MOPS version which was breaking things and other MISC issues
Though the gen-tests still require haskell/cabal which was nice with NIX, the CI just installed it manually anyways
I don't know if you have strong thoughts on it. Im not skilled in NIX and I just quit after a while of not being able to fix it

nomeata commented:
Looks like I wrangled `npm` enough:
```
~/dfinity/ic-certification $ npx mops owner add 7357l-qewss-qxuo3-ccvfm-6ofer-bameo-jmkvh-kzlxv-vs4w7-zauoh-jae
✔ Are you sure you want to add owner 7357l-qewss-qxuo3-ccvfm-6ofer-bameo-jmkvh-kzlxv-vs4w7-zauoh-jae to ic-certification package? … yes
Success!
Added owner 7357l-qewss-qxuo3-ccvfm-6ofer-bameo-jmkvh-kzlxv-vs4w7-zauoh-jae to package ic-certification
```
